### ISSUE TITLE
Importer v2: Required fields

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1614]
As previously discussed, I want to make the "Stage (extensive)" attribute mandatory for records imported into the iRecord Import survey. I think I can do that by adjusting the attribute settings for the survey via the warehouse interface.

If I do that:
1. Will it automatically be picked up in the list of tick-boxes within the importer that show whether all required fields have been matched?
2. And are there any other issues to consider before making this change? I'm aware that it will mean we have a lot of previously imported records that are missing the stage, and that if any of those records require editing a stage term would then have to be added before saving them, but I think that is acceptable.

**Comment by DavidHepper** [comment]
How would it interact with those of us having a list of Life Stage Abundance custom fields instead of a single Stage/Count pair? Or do you mean just for that Survey Type?

**Comment by kitenetter** [comment]
@DavidHepper my understanding is that this is specific to the chosen survey, so making a change to iRecord Import will not affect the dragonfly import survey.

**Comment by kitenetter** [comment]
@johnvanbreda yesterday I set the survey-specific settings for the "iRecord Import" survey so that "Stage (extensive)" is a mandatory attribute for that survey. However, it is not being displayed in the list of "Required fields" in the "Map import columns" section of the v2 importer.

I also checked the "iRecord Moths" survey, which has always has a survey-specific setting for Stage to be mandatory, and that is also not showing in the "Required fields" list.

Is there something else we need to do for this to be handled correctly?

**Comment by kitenetter** [comment]
@johnvanbreda further to this, I've just tried importing a record into "iRecord Import". As per the previous comment, the Stage attribute is not listed as required on the mapping page, but when I tried to import a record with no column for stage the record failed at the final import step.

This produced an error file, which contained the message:
`{"unknown column": "A value for this attribute is required."}`

Ideally the mapping page should automatically list all attributes that are required, and then any subsequent error message should include information about which attribute is missing.
